spec/openapi.yaml

```diff
@@ -120,6 +120,39 @@ paths:
         '500':
           description: An unexpected error occured.
 
+  /tag/metadata/delete:
+    post:
+      operationId: tag-metadata-delete
+      summary: Permanently deletes the specified metadata from an existing tag.
+      description: >
+        Provide the metadata names that you want to delete.
+      tags:
+        - Tags
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/TagMetadataDeleteRequest'
+      responses:
+        '200':
+          description: Tag metadata successfully deleted.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TagMetadataDeleteResponse'
+        '204':
+          description: >
+            (No Content) when there is no need to return any content, where metadata was not found or if it was already deleted.
```

Comment:
Sorry to say but this design is still problematic, as we allow multiple `metadataNames` to be provided, but only allow for a unary response.  I suggest we discuss to find clean solution.